### ISSUE TITLE
Various Token Issues

### ISSUE THREAD

**Issue by azdraik** [magefree/mage#4533]
Betrayers of Kamigawa has two spirt tokens.  One is a White 3/3 Flying spirit and the other is a Colorless 1/1 spirit.  The image pathway is the same for both images.  TOK.zip\BOK\Spirit.full.jpg

Lorwyn has 2 Elemental Tokens. One is a White 4/4 Elemental and the other is a Green 4/4 Elemental.  The image pathway is the same for both images.  TOK.zip\LRW\Elemental.full.jpg

Shadowmoor has 2 Elf Warrior.  One is a Green 1/1 Elf Warrior and the other is a White and Green 1/1 Elf Warrior.  The image pathway is the same for both images.  TOK.zip\SHM\Elf Warrior.full.jpg

Zendikar has 2 Elemental Tokens.  One is a Red 3/1 Elemental and the other is a Red 7/1 Elemental with Haste.  The image pathway is the same for both images.  TOK.zip\ZEN\Elemental.full.jpg

Duel Decks: Phyrexia vs. the Coalition has a duplicate entry for the Hornet token when reviewing the tokens in the Viewer.  There should only be one Hornet token entry. TOK.zip\DDE\Hornet.full.jpg

Archenemy has 2 goblin tokens.  One is a Red 1/1 Goblin and the other is a Red 2/1 Goblin with haste.  The image pathway is the same for both images.  TOK.zip\ARC\Goblin.full.jpg

Magic 2011 appears to be missing the Green 1/1 Ooze token which is created by the death of the Green 2/2 Ooze token, there is no 1/1 Ooze token in the viewer.

New Phyrexia has 2 insect tokens listed in the viewer.  Based on the set I believe there should only be Green 1/1 Insect with Infect listed.  I was unable to locate anything in that set which makes Green 1/1 Insect (with no keywords) tokens. 

Commander (CMD) has 2 beast tokens.  One is a Green 4/4 and the other is a Green 3/3.  The Image pathway is the same for both tokens.  TOK.zip\CMD\Beast.full.jpg

Innistrad has 2 wolf tokens.  In the viewer currently both show as a Green 2/2 wolf.  One of those should be a Black 1/1 wolf with deathtouch.  As both appear to be coded as Green 2/2 wolf tokens unsure which of these should be the Black 1/1 Wolf with deathtouch

Avacyn Restored has 2 spirit tokens.   In the viewer there is a White 1/1 Spirit with flying and a Colorless 1/1 Spirit.  I believe the Colorless 1/1 Spirit should be a Blue 1/1 Spirit with flying.  Current file path for the colorless spirit is TOK.zip\AVR\Spirit 2.full.jpg

Return to Ravnica has 1 elemental token.  In the viewer it is showing as a Red 3/1 Elemental token.  There should be a Green and White 8/8 Elemental with Vigilance token.  TOK.zip\RTR\Elemental.full.zip

Updates added 3-5-18
Commander 2013 has 3 beast tokens that share a file path.  There is a Green 4/4 Beast, Green 5/5 Beast and a Green 5/5 Beast with Trample.  TOK.zip\C13\Beast.full.jpg
There are also 3 different elemental tokens which share a file path.  A Green 1/1 Elemental, a Green 4/4 Elemental and a Red 1/1 Elemental with Haste. TOK.zip\C13\Elemental.full.jpg

Duel Decks: Anthology, Garruk vs Liliana has 2 beast tokens which share a file path. There is a Green 3/3 Beast and a Green 4/4 Beast. TOK.zip\DD3GVL\Beast.full.jpg

Commander 2016 has 2 bird tokens.  The viewer shows both of them being a White 1/1 Bird.  This is incorrect one should be a White 1/1 Bird and one should be a Blue 1/1 Bird.  
The elemental also appears to be incorrect, the viewer lists a Red 3/1 Elemental while the token should be a White 4/4 Elemental.
There are 2 spirit tokens they are both listed as Colorless 1/1 Spirits when there should be a Colorless 1/1 Spirit and a White 1/1 Spirit.

Modern Masters 2017 there are 2 beast tokens both are listed in the viewer as Green 3/3 Beasts while it should be a Green 3/3 Beast and Green 4/4 Beast. 
There are 2 soldier tokens both listed as a White 1/1 Soldier.  There should be a White 1/1 Soldier and a Red and White 1/1 Soldier.
The spider token is listed as a Green 1/3 Spider with Reach when is should be a Black 2/4 with Reach.

**Comment by spjspj** [comment]
You can make images such as the following in TOK.zip.  It's the alternate naming scheme which describes what type of token the game is after.  It will look for these types of images after an initial fail oto find say TOK.zip/AVR/Spirit 2.full.jpg:
```
ANGEL.B.ANGEL.CREATURE.3.3.full.jpg
ANGEL.W.ANGEL.CREATURE.3.3.full.jpg
ANGEL.W.ANGEL.CREATURE.4.4.full.jpg
APE.G.APE.CREATURE.2.2.full.jpg
APE.G.APE.CREATURE.3.3.full.jpg
ASHAYATHEAWOKENWORLD.G.ELEMENTAL.CREATURE.4.4.full.jpg
ASSASSIN.B.ASSASSIN.CREATURE.1.1.full.jpg
ASSEMBLYWORKER..ASSEMBLYWORKER.ARTIFACTCREATURE.2.2.full.jpg
AVATAR.W.AVATAR.CREATURE.S.S.full.jpg
BAT.B.BAT.CREATURE.1.1.full.jpg
BAT.B.BAT.CREATURE.1.2.full.jpg
BEAR.G.BEAR.CREATURE.2.2.full.jpg
BEAR.G.BEAR.CREATURE.4.4.full.jpg
BEAST..BEAST.ARTIFACTCREATURE.6.6.full.jpg
BEAST.B.BEAST.CREATURE.3.3.full.jpg
BEAST.G.BEAST.CREATURE.2.2.full.jpg
BEAST.G.BEAST.CREATURE.3.3.full.jpg
BEAST.G.BEAST.CREATURE.4.2.full.jpg
BEAST.G.BEAST.CREATURE.4.4.full.jpg
BEAST.G.BEAST.CREATURE.5.5.full.jpg
BEAST.WRG.BEAST.CREATURE.8.8.full.jpg
BIRD.R.BIRD.CREATURE.4.4.full.jpg
BIRD.U.BIRD.CREATURE.1.1.full.jpg
BIRD.U.BIRD.CREATURE.2.2.full.jpg
BIRD.U.BIRD.CREATUREENCHANTMENT.2.2.full.jpg
BIRD.W.BIRD.CREATURE.1.1.full.jpg
BIRD.W.BIRD.CREATURE.3.3.full.jpg
BIRD.W.BIRD.CREATURE.3.4.full.jpg
BIRD.WU.BIRD.CREATURE.1.1.full.jpg
BIRDSOLDIER.W.BIRD.CREATURE.1.1.full.jpg
BOAR.G.BOAR.CREATURE.2.2.full.jpg
BOAR.G.BOAR.CREATURE.3.3.full.jpg
BRAINIAC.R.BRAINIAC.CREATURE.1.1.full.jpg
BUTTERFLY.G.INSECT.CREATURE.1.1.full.jpg
CAMARID.U.CAMARID.CREATURE.1.1.full.jpg
CARIBOU.W.CARIBOU.CREATURE.0.1.full.jpg
CARNIVORE.R.BEAST.CREATURE.3.1.full.jpg
CAT.B.CAT.CREATURE.2.1.full.jpg
CAT.G.CAT.CREATURE.1.1.full.jpg
CAT.R.CAT.CREATURE.1.1.full.jpg
CAT.W.CAT.CREATURE.1.1.full.jpg
CAT.W.CAT.CREATURE.2.2.full.jpg
CATSOLDIER.W.CATSOLDIER.CREATURE.1.1.full.jpg
CATWARRIOR.G.CATWARRIOR.CREATURE.2.2.full.jpg
CENTAUR.G.CENTAUR.CREATURE.3.3.full.jpg
CENTAUR.G.CENTAUR.CREATURE.3.3a.full.jpg
CENTAUR.G.CENTAUR.CREATUREENCHANTMENT.3.3.full.jpg
CITIZEN.W.CITIZEN.CREATURE.1.1.full.jpg
CLERIC.W.CLERIC.CREATUREENCHANTMENT.2.1.full.jpg
CLERIC.WB.CLERIC.CREATURE.1.1.full.jpg
CLOUDSPRITE.U.FAERIE.CREATURE.1.1.full.jpg
CLUE..CLUE.ARTIFACT.0.0.full.jpg
CONSTRUCT..CONSTRUCT.ARTIFACTCREATURE.1.1.full.jpg
CONSTRUCT..CONSTRUCT.ARTIFACTCREATURE.6.12.full.jpg
DEMON.B.DEMON.CREATURE.5.5.full.jpg
DEMON.B.DEMON.CREATURE.S.S.full.jpg
DEVIL.R.DEVIL.CREATURE.1.1.full.jpg
etc
```

**Comment by azdraik** [comment]
I understand that is an option.  Was just reporting issues I found while going through tok.zip on a set by set basis since I did not care for all the star city tokens.

**Comment by Zerrisx** [comment]
I think some of these were fixed with aeef4e1.  Is there an easy way to test what tokens link to what images, so I could check for myself?

**Comment by JayDi85** [comment]
@Zerrisx How to find bug like "wrong token image in game" (fix example aeef4e1193b786727ebba915f4eefdcc6b6b423a) -- I think it can be automated by some verify unit test too:
1. Open `card-pictures-tok.txt`;
2. Look at tokens list and find duplicated names in one set:
![shot_200707_221804](https://user-images.githubusercontent.com/8344157/86824844-be1b1380-c09f-11ea-85b7-7379aac3f750.png)
3. Each duplicated token record must have:
  * Token index (example: 1, 2, 3);
    * Possible bug on missing index: xmage will put on battlefield only one token image instead random;
  * Token class (example: `AngelTombToken`);
    * Possible bug on missing class name: xmage will put on battlefield wrong image with different characteristics;
4. Find tokens with index 2;
5. Open class token file (if you see only name then add `Token` to that name: `Angel` -> `AngelToken.java`);
6. If token from `card-pictures-tok.txt` have index 2+ then it must be coded with `setTokenType` :
![shot_200707_223234](https://user-images.githubusercontent.com/8344157/86826415-c4aa8a80-c0a1-11ea-8cb6-5bca99f12f95.png)
![shot_200707_223527](https://user-images.githubusercontent.com/8344157/86826718-2cf96c00-c0a2-11ea-90c8-7cc65ac223b2.png)
7. If it don't have that code then it 100% bugged (game put wrong image for that token -- from index 1; or uses same token image instead random);

**Comment by PurpleCrowbar** [comment]
Commenting to self-assign this.
